Rework Introduction and Overview of Project Syn (#51)
projectsyn/documentation: docs/modules/ROOT/pages/architecture.adoc

            \-----+------+--------+--/                    +------------------+  |   +--------------------+            |
                  ^      ^        ^                                             |                                     |
                  |      |        |                                             |                                     |
          +-------+      +-----+  +--------------------+                        |                                     |
          |                    |                       |                        |                                     |
+---------------------------------------------+        |                        |                                     |
|         |                    |              |        |                        |                                     |
| +-------+-------+ +----------+----------+   |  +-----+------+                 |                                     |
| |               | |                     |   |  |            |                 |                                     |
| | Common        | | Tenant and cluster  |   |  | Commodore  |                 |                                     |

> | | configuration | | configuration       |   |  | Components |                 |                                     |

Reviewer: make all 3 git repos (common, tentant, catalog) the same color and/or include "Git repo" in all three boxes?

| |               | |                     |   |  |            |                 |                                     |
| +---------------+ +---------------------+   |  +-----+------+                 |                                     |
|                   Git repo managed by       |        ^                        |                                     |

> |                   Operator                  |        | Update                 |                                     |

Reviewer: Git repo created & configured by _Lieutenant_ Operator?

(I think the word "manage" is very unspecific :)

+------------------------------------+--------+        |                        +-------------------------------------+
Configuration Git repositories       ^                 |
                                     | Update          |
                                     |                 |
                           +---------+------------+    |
                           |                      |    |
                           | Renovate             +----+
                           |                      |
                           +----------------------+
....

Project Syn main tools:

[cols=2*,options=header]
|===
|Tool
|Purpose

|xref:commodore:ROOT:index.adoc[Commodore]
|Configuration generator leveraging a hierarchical configuration model. Uses https://kapitan.dev/[Kapitan] under the hood.

|xref:lieutenant-operator:ROOT:index.adoc[Lieutenant Operator]

> |Implements the business logic for handling all aspects around Git repositories, clusters and tenants. Manages the CRDs `Cluster`, `Tenant`, `GitRepo`.

Reviewer: ```suggestion
|Manages the CRDs `Cluster`, `Tenant`, `GitRepo`. Implements the business logic for handling all aspects around Git repositories for clusters and tenants. 
```

|xref:lieutenant-api:ROOT:index.adoc[Lieutenant API]

> |REST API to expose the functionality of the Lieutenant API and to enforce some defaults.

Reviewer: ```suggestion
|REST API to expose the functionality of the Lieutenant and to enforce some defaults.
```

|xref:steward:ROOT:index.adoc[Steward]

> |In-cluster agent which bootstraps and manages the GitOps operator Argo CD and takes care of keeping the Syn cluster tools up and running.

Reviewer: ```suggestion
|In-cluster agent which bootstraps, configures and starts Argo CD (GitOps operator) and takes care of keeping the Syn cluster tools up and running.
```

                           |                      |                             | Git repository cGRE    |
                           +----------------------+                             +------------------------+
....

Project Syn main tools:

[cols=2*,options=header]
|===
|Tool
|Purpose

|xref:commodore:ROOT:index.adoc[Commodore]
|Configuration generator leveraging a hierarchical configuration model. Uses https://kapitan.dev/[Kapitan] under the hood.

|xref:lieutenant-operator:ROOT:index.adoc[Lieutenant Operator]
|Manages the CRDs `Cluster`, `Tenant`, `GitRepo`. Implements the business logic for handling all aspects around Git repositories for clusters and tenants. 

|xref:lieutenant-api:ROOT:index.adoc[Lieutenant API]
|REST API to expose the functionality of the Lieutenant Operator and to enforce some defaults.

|xref:steward:ROOT:index.adoc[Steward]
|In-cluster agent which bootstraps, configures and starts Argo CD (GitOps operator) and takes care of keeping the Syn cluster tools up and running.

|===

Third-party tools used in Project Syn:

[cols=2*,options=header]
|===
|Tool
|Purpose

|https://argoproj.github.io/argo-cd/[Argo CD]

> |tbd

Reviewer 2: I guess this is one of the reasons why the PR is still in draft.
